test/integration/spec/rtcpeerconnection.js

```diff
@@ -786,42 +735,133 @@ function makeStream(constraints) {
   return new Promise((resolve, reject) => getUserMedia(resolve, reject));
 }
 
-function testAddStream() {
+function testAddTrack() {
   var test;
   var stream;
 
-  before(() => {
-    return makeStream().then(_stream => stream = _stream);
+  before(async () => {
+    stream = await makeStream();
   });
 
-  beforeEach(() => {
-    return makeTest().then(_test => test = _test);
+  beforeEach(async () => {
+    test = await makeTest();
+  });
+
+  context('when the RTCPeerConnection is closed', () => {
+    var exception;
+
+    beforeEach(() => {
+      test.peerConnection.close();
+      try {
+        test.peerConnection.addTrack(stream.getTracks()[0], stream);
+      } catch (e) {
+        exception = e;
+      }
+    });
+
+    it('should throw', () => {
+      assert(!!exception);
+    });
+
+    it('should not add the MediaStreamTrack', () => {
+      const addedTracks = new Set('getSenders' in RTCPeerConnection.prototype
```

Comment:
Aren't we shimming `getSenders` all the time now? So we don't need this `in` check?

Reply:
@markandrus I did not want to depend on another shim in this test. I can change it if you think its OK to use another shim while testing the `addTrack` shim.

Reply:
I don't understand: in these tests, we are testing our shims, and so `getSenders` should be available, right?